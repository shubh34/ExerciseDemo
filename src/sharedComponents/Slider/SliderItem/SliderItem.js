import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './SliderItem.scss';
import Scores from '../../Scores/Scores';

const SliderItem = ({
	movie, id, score, votes
}) => (
	<div className='slider-item'>
		<Link className="slider-item-image" to={`/movies/${id}`}>
			<img width="100%" src={movie} alt="" />
		</Link>
		<Scores score={score} votes={votes} />
	</div>
);

SliderItem.propTypes = {
	movie: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
	score: PropTypes.number.isRequired,
	votes: PropTypes.string.isRequired
};

export default SliderItem;
