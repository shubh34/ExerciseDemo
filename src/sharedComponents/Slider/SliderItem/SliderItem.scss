.slider-item {
    flex: 0 0 35.5%;
    padding: 0 5px;
    position: relative;
    border-radius: 2px;
    @media screen and (min-width: 560px) {
      flex: 0 0 22.5%;
    }
    @media screen and (min-width: 1140px) {
      flex: 0 0 15.5%;
    }
    @media screen and (min-width: 1400px) {
      flex: 0 0 13.5%;
    }

  
   
  }
  .slider-item-image {
    vertical-align: top;
  }
  .slider-item-image :hover {
    border: 3px #ffe300 solid;
    padding: 0px;
    cursor: pointer;
    border-radius: 5px; 
  }
