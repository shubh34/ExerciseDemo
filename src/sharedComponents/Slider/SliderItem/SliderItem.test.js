

import React from 'react';
import { shallow } from 'enzyme';

import SliderItem from './SliderItem';


const setUp = (container = shallow) => {
	const props = {
		id: '1',
		movie: 'https://images-2.wuaki.tv/system/artworks/86665/master/capitana-marvel-1560352921.jpeg',
		score: 22,
		votes: '6.6k'
	};
	const wrapper = container(<SliderItem {...props} />);
	return { wrapper };
};

describe('SliderItem', () => {
	it('should render SliderItem', () => {
		const { wrapper } = setUp();
		expect(wrapper).toMatchSnapshot();
	});
});
