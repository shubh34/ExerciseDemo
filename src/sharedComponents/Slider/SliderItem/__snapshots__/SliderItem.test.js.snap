// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItem should render SliderItem 1`] = `
<div
  className="slider-item"
>
  <Link
    className="slider-item-image"
    to="/movies/1"
  >
    <img
      alt=""
      src="https://images-2.wuaki.tv/system/artworks/86665/master/capitana-marvel-1560352921.jpeg"
      width="100%"
    />
  </Link>
  <Scores
    score={22}
    votes="6.6k"
  />
</div>
`;
